src/test/java/data/exceptions/SetPriorityExceptionTest.java

```diff
@@ -0,0 +1,147 @@
+        String[] dummyTestTimes = new String[]{null};
+        addTask(date, taskDescription, testTaskType, dummyTestDates, dummyTestTimes);
+        List<Task> dayTasks = getDayTasks(date);
+
+        // Assert
+        assertDoesNotThrow(() -> checkIfTaskIndexIsValidForPriority(dayTasks, 1));
+
+    }
+
+    @Test
+    public void checkIfTaskIndexIsValidForPriority_invalidIndexGivenForDayWithTask_exceptionThrown ()
+            throws TaskManagerException {
+
+        // Arrange
+        LocalDate date = LocalDate.now();
+        String taskDescription = "Test Todo";
+
+        // Act
+        TaskType testTaskType = TaskType.TODO;
+        String[] dummyTestDates = new String[]{null};
+        String[] dummyTestTimes = new String[]{null};
+        addTask(date, taskDescription, testTaskType, dummyTestDates, dummyTestTimes);
+        List<Task> dayTasks = getDayTasks(date);
+
+        // Assert
+        SetPriorityException thrown = assertThrows(SetPriorityException.class, () ->
+                checkIfTaskIndexIsValidForPriority(dayTasks, 2));
+
+        assertEquals("The task index you attempted to set a priority to is out of range!", thrown.getMessage());
+
+    }
+
+    @Test
+    public void checkIfTaskIndexIsValidForPriority_indexGivenForDayWithNoTasks_exceptionThrown () {
+
+        // Arrange
+        LocalDate date = LocalDate.now();
+
+        // Act
+        List<Task> dayTasks = getDayTasks(date);
+
+        // Assert
+        SetPriorityException thrown = assertThrows(SetPriorityException.class, () ->
+                checkIfTaskIndexIsValidForPriority(dayTasks, 1));
+
+        assertEquals("There are no tasks to set a priority to on this day!", thrown.getMessage());
+
+    }
+
+    @Test
+    public void checkIfPriorityIsValid_validHPriorityGiven_noExceptionThrown () {
+        // Arrange
+        String priorityString = "H";
+
+        // Act and Assert
+        assertDoesNotThrow(() -> checkIfPriorityIsValid(priorityString));
+    }
+
+    @Test
+    public void checkIfPriorityIsValid_validMPriorityGiven_noExceptionThrown () {
+        // Arrange
+        String priorityString = "M";
+
+        // Act and Assert
+        assertDoesNotThrow(() -> checkIfPriorityIsValid(priorityString));
+    }
+
+    @Test
+    public void checkIfPriorityIsValid_validLPriorityGiven_noExceptionThrown () {
+        // Arrange
+        String priorityString = "L";
+
+        // Act and Assert
+        assertDoesNotThrow(() -> checkIfPriorityIsValid(priorityString));
+    }
+
+    @Test
+    public void checkIfPriorityIsValid_invalidPriorityGiven_exceptionThrown () {
+        // Arrange
+        String priorityString = "low";
+
+        // Act and Assert
+        SetPriorityException thrown = assertThrows(SetPriorityException.class, () ->
+                checkIfPriorityIsValid(priorityString));
+        assertEquals("The priority you entered is invalid! Please enter a valid priority (L, M, H)!",
+                thrown.getMessage());
```

Comment:
Great job adding in so many more test cases! Really makes everything more robust.